control-plane/agents/core/src/core/states.rs

```diff
@@ -0,0 +1,88 @@
+use common_lib::types::v0::{
+    message_bus::{Nexus, NexusId, Pool, PoolId, Replica, ReplicaId},
+    store::{nexus::NexusState, pool::PoolState, replica::ReplicaState},
+};
+use std::{collections::HashMap, ops::Deref, sync::Arc};
+use tokio::sync::{Mutex, RwLock};
+
+/// Locked Resource States
+#[derive(Default, Clone, Debug)]
+pub(crate) struct ResourceStatesLocked(Arc<RwLock<ResourceStates>>);
+
+impl ResourceStatesLocked {
+    pub(crate) fn new() -> Self {
+        ResourceStatesLocked::default()
+    }
+}
+
+impl Deref for ResourceStatesLocked {
+    type Target = Arc<RwLock<ResourceStates>>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Resource States
+#[derive(Default, Debug)]
+pub(crate) struct ResourceStates {
+    /// Todo: Add runtime state information for nodes.
+    pub(crate) nexuses: HashMap<NexusId, Arc<Mutex<NexusState>>>,
+    pub(crate) pools: HashMap<PoolId, Arc<Mutex<PoolState>>>,
```

Comment:
Do these things have to public? Even within a crate i would *default* them to NOT be public. It think having code that returns an iterator would be better. Having foo.bar.field access makes code very very hard to refactor.

Reply:
I have changed things so that these are now private.